src/Microsoft.AspNet.Server.Kestrel/Http/SocketOutput.cs

```diff
@@ -16,11 +16,13 @@ public class SocketOutput : ISocketOutput
     {
         public const int MaxPooledWriteReqs = 1024;
 
-        private const int _maxPendingWrites = 3;
```

Comment:
It was @lodejard who initially suggested this pattern. I think that it had something to do about prioritizing future writes even if there was a pending write operation ready to handle the newly requested write.

I'm not sure I fully understood the explanation for having up to 3 pending write requests, because only having one queued write at a time still makes sense to me. I'll talk to him about it.